### PR TITLE
3867 fix non intensity error

### LINKED ISSUE #3867: `RandKSpaceSpikeNoise` got an unexpected keyword argument 'dim'
**Describe the bug**
`RandKSpaceSpikeNoise` does not work with NumPy array if `intensity_range` is not set manually.

In case a NumPy array is provided, the linked part will call `np.mean(.., dim=...)` but NumPy does not support dim arguments (should be `axis`). The implementation works fine for `torch.Tensors`.

https://github.com/Project-MONAI/MONAI/blob/d6327a765cf23dafd680b2a4b472edf9639e1087/monai/transforms/intensity/array.py#L1816-L1818

**To Reproduce**
Call transform on NumPy array with default values (specifically `intensity_range =None`).

**Expected behavior**
No crash.

**Screenshots**

**Environment**

Ensuring you use the relevant python executable, please paste the output of:

```
python -c 'import monai; monai.config.print_debug_info()'
```

**Additional context**

### PR DESCRIPTION
**Pull request by yiheng-wang-nv**
Signed-off-by: Yiheng Wang <vennw@nvidia.com>

Fixes #3867 .

### Description
This PR fixes the `_set_default_range` issue in class `RandKSpaceSpikeNoise`.

There are two errors in this function:

1. For numpy based img, `np.mean` does not have the arg `dim` (this bug is reported in #3867 , thanks to @mibaumgartner !)
2. the generated intensity range has the same type as `img`, if `img` is saved in GPU, error will be raised in the following `self.R.uniform` manipulation step.

In addition, this PR adds the test case (none intensity_range) for `RandKSpaceSpikeNoise`, and the test case (none k_intensity) for `KSpaceSpikeNoise`.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/transforms/intensity/array.py [hunk 1]
```diff
@@ -1813,9 +1813,10 @@ def _set_default_range(self, img: NdarrayOrTensor) -> Sequence[Sequence[float]]:
         n_dims = len(img.shape[1:])
 
         k = self.shift_fourier(img, n_dims)
+        k, *_ = convert_data_type(k, output_type=np.ndarray)
```
Comment (Nic-Ma) [inline monai/transforms/intensity/array.py:1816]: May I know why convert `k` to numpy array here?

Thanks.

Comment (yiheng-wang-nv) [reply monai/transforms/intensity/array.py:1816]: If k is torch.tensor in gpu device, this line: https://github.com/Project-MONAI/MONAI/pull/3871/files#diff-83ae933a39467d71ab978c0bba9debaa2db45f24a74c127a664f1044fc0017dfR1784 will raise an error.

In addition, I think the generated intensity range should has the same type as the argument `intensity_range`:
Sequence[Union[Sequence[float], float]]

**Review by Nic-Ma — APPROVED** [review]
Hi @rijobro @wyli ,

Looks good to me, but I am not very familiar with the algorithm here, could you please help double confirm it?

Thanks in advance.

File: monai/transforms/intensity/array.py (hunk 1 above)
Comment (yiheng-wang-nv) [reply monai/transforms/intensity/array.py]: Thanks @Nic-Ma , I submitted a new commit, and put the convert step after calculation.
